### ISSUE TITLE
[web:a11y] fall back isSelected to aria-current

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#167147]
`aria-selected` only supports gridcell, option, row and tab roles ([source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-selected)). Flutter supports row and tab, and may in the future support gridcell and option. However, some widgets, such as `RawChip`, use the `isSelected` flag outside of these roles. This flag may also appear in downstream widgets that the Flutter framework does not have control over. When this happens, screen readers on the web would ignore `aria-selected` and provide no guidance to the user w.r.t. to the state of the widget.

One way to deal with this is to restrict `isSelected` flag to specific roles known to support it on the web. That would work if this flag was introduced just now, but `isSelected` has existed since before we added `aria-selected` support. The new restriction would be a severe breaking change. Additionally, mobile supports the selected property without role restrictions, so there are plenty of widgets in the wild that use it, and we want to support those widgets on the web as well as we can.

## Proposal

When rendering `SemanticsFlag.isSelected` the web engine will first check the role assigned to the respective semantics node. If it's one of the roles supported by `aria-selected`, use the `aria-selected` attribute. Otherwise, fall back to `aria-current`. While `aria-current` means something slightly different, it is pretty close in meaning, and it does not have the role restriction. Using it would be better for user experience than having the browser ignore `aria-selected` entirely.
